lms/static/js/bookmarks/collections/bookmarks.js

```diff
@@ -1,20 +1,24 @@
 ;(function (define) {
     'use strict';
-    define(['backbone', 'js/bookmarks/models/bookmark'],
-        function (Backbone, BookmarkModel) {
+    define(['backbone', 'common/js/components/collections/paging_collection', 'js/bookmarks/models/bookmark'],
+        function (Backbone, PagingCollection, BookmarkModel) {
 
+            return PagingCollection.extend({
+                initialize: function(options) {
+                    PagingCollection.prototype.initialize.call(this);
 
-        return Backbone.Collection.extend({
-            model: BookmarkModel,
+                    this.course_id = options.course_id;
+                    this.server_api.course_id = function () { return encodeURIComponent(this.course_id); };
+                    this.server_api.fields = function () { return encodeURIComponent('display_name,path'); };
```

Comment:
`display_name`: Display name of the XBlock.	
`path`: List of dicts containing {"usage_id": "", display_name:""} for the XBlocks from the top of the course tree till the parent of the bookmarked XBlock.